packages/uploadthing/test/upload-builder.test.ts

```diff
@@ -125,6 +156,21 @@ it("with optional input", () => {
       expectTypeOf<{ foo: string } | undefined>(opts.input);
       return {};
     });
+
+  f(["image"])
+    .input(v.optional(v.object({ foo: v.string() })))
+    .middleware((opts) => {
+      expectTypeOf<{ foo: string } | undefined>(opts.input);
+      return {};
+    });
+
+  // FIXME
+  // f(["image"])
+  //   .input(Schema.Struct({ foo: Schema.String }).pipe(Schema.optional))
+  //   .middleware((opts) => {
+  //     expectTypeOf<{ readonly foo: string } | undefined>(opts.input);
+  //     return {};
+  //   });
```

Comment:
hmm getting this to work seems to require more types than i anticipated... I thought `Schema.Schema` was the base type, but when using optional the type is wrapped like `optional<Schema.Schema>`